src/header/library.cairo

```diff
@@ -193,13 +200,50 @@ namespace BlockHeaderVerifier:
     }(ctx : BlockHeaderValidationContext):
         alloc_locals
         # Write current header to storage
-        block_header_hash_.write(ctx.height, ctx.block_header.hash)
-        block_header_.write(ctx.block_header.hash, ctx.block_header)
+        write_header_to_storage(ctx.height, ctx.block_header)
 
         # Consensus rules callback
         previous_block_hash.on_block_accepted(ctx)
         check_pow.on_block_accepted(ctx)
         median_past_time.on_block_accepted(ctx)
         return ()
     end
+
+    func write_header_to_storage{
+        syscall_ptr : felt*,
+        pedersen_ptr : HashBuiltin*,
+        range_check_ptr,
+        bitwise_ptr : BitwiseBuiltin*,
+    }(height : felt, block_header : BlockHeader):
+        block_header_hash_.write(height, block_header.hash)
+        block_header_.write(block_header.hash, block_header)
+        return ()
+    end
+
+    # Returns the hardcoded genesis block.
+    # See https://github.com/bitcoin/bitcoin/blob/master/src/chainparams.cpp
+    # and https://www.blockchain.com/btc/block/000000000019d6689c085ae165831e934ff763ae46a2a6c172b3f1b60a8ce26f
+    func genesis_block_header{range_check_ptr}(genesis_block_header : BlockHeader):
+        let (genesis_hash : Uint256) = felt_to_uint256(
+            0x19d6689c085ae165831e934ff763ae46a2a6c172b3f1b60a8ce26f
+        )
+        let (genesis_merkel_root : Uint256) = felt_to_uint256(
+            0x4a5e1e4baab89f3a32518a88c31bc87f618f76673e2cc77ab2127b7afdeda33b
+        )
+        let genesis_block_header = BlockHeader(
+            hash=genesis_hash,
+            version=1,
+            timestamp=1231006505,
+            nonce=2083236893,
+            bits=0x1d00ffff,
+            prev_block=Uint256(0, 0),
+            merkle_root=genesis_merkel_root,
+        )
+        return (genesis_block_header)
+    end
+
+    func felt_to_uint256{range_check_ptr}(x) -> (res : Uint256):
+        let (hi, lo) = split_felt(x)
+        return (Uint256(lo, hi))
```

Comment:
This function should  be imported from another library after refactor, right?

Reply:
yes indeed, the refactor will come with my PR about target rule